internal/controller/policyrecommendation_registrar.go

```diff
@@ -27,55 +29,71 @@ type PolicyRecommendationRegistrar struct {
 	Scheme *runtime.Scheme
 }
 
-func (r *PolicyRecommendationRegistrar) Reconcile(ctx context.Context,
-	request ctrl.Request) (ctrl.Result, error) {
+func (r *PolicyRecommendationRegistrar) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger = logger.WithValues("request", request)
 
-	log := log.FromContext(ctx)
+	// Check if Rollout exists
+	rollout := argov1alpha1.Rollout{}
+	err := r.Client.Get(ctx, request.NamespacedName, &rollout)
+	if err == nil {
+		// Rollout exists, create policy recommendation
+		return r.createPolicyRecommendation(ctx, &rollout, logger)
+	}
 
-	// Fetch the Deployment or Rollout instance
-	//TODO handle the case for deployment
-	instance := argov1alpha1.Rollout{}
-	err := r.Client.Get(ctx, request.NamespacedName, &instance)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			log.Info("Object not found, could have been deleted after reconcile request")
-			// Object not found, could have been deleted after reconcile request.
-			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
-			// Return and don't requeue
-			return reconcile.Result{}, nil
-		}
-		// Error reading the object - requeue the request.
-		log.Error(err, "Error reading the object - requeue the request")
+	if !errors.IsNotFound(err) {
+		// Error occurred
+		logger.Error(err, "Failed to get Rollout. Requeue the request")
+		return reconcile.Result{}, err
+	}
+
+	// Check if Deployment exists
+	deployment := appsv1.Deployment{}
+	err = r.Client.Get(ctx, request.NamespacedName, &deployment)
+	if err == nil {
+		// Deployment exists, create policy recommendation
+		return r.createPolicyRecommendation(ctx, &deployment, logger)
+	}
+
+	if !errors.IsNotFound(err) {
+		logger.Error(err, "Failed to get Deployment. Requeue the request")
 		return reconcile.Result{}, err
 	}
 
+	logger.Info("Rollout or Deployment not found. It could have been deleted.")
+	return reconcile.Result{}, nil
```

Comment:
nit: reconcile.Result{} and ctrl.Result{} are being used since both are same. However we can stick to one convention for uniformity. I had used ctrl.Result{} in my previous controller implementations.

Reply:
Have moved everything to ctrl.Result.